### PR TITLE
add Dockerfile for building release wheels

### PR DESCRIPTION
**Pull request by mavenlin**
Based on ubuntu16 image.

### CODE REVIEW

File: Dockerfile-build [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM ubuntu:16.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN sed -i "s/archive.ubuntu.com/mirror.0x.sg/g" /etc/apt/sources.list
```
Comment (zhongwen) [inline Dockerfile-build:5]: might not use `sg` ?

Comment (mavenlin) [reply Dockerfile-build]: Good point, removed

**Comment by Trinkle23897** [comment]
https://github.com/pypa/cibuildwheel

File: docker/release.dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# install base dependencies
+
+RUN apt-get update && apt-get install -y software-properties-common && add-apt-repository ppa:ubuntu-toolchain-r/test && add-apt-repository ppa:deadsnakes/ppa
+RUN apt-get update \
+    && apt-get install -y git curl wget gcc-9 g++-9 build-essential patchelf make libssl-dev zlib1g-dev \
+    libbz2-dev libreadline-dev libsqlite3-dev llvm \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev \
+    python3.6 python3.7 python3.8 python3.9 python3.10 \
+    python3.6-dev python3.7-dev python3.8-dev python3.9-dev python3.10-dev \
```
Comment (zhongwen) [inline docker/release.dockerfile:17]: `python3.6` is dropped right?

Comment (Trinkle23897) [reply docker/release.dockerfile:17]: yep but it doesn't matter to build wheels

**Review by Trinkle23897 — APPROVED** [review]

**Pull request merged**